python/ray/tune/utils/util.py

```diff
@@ -298,6 +298,32 @@ def unflatten_dict(dt, delimiter="/"):
     return out
 
 
+def unflatten_list_dict(dt, delimiter="/"):
+    """Unflatten nested dict and list."""
+    out_type = list if list(dt)[0].split(delimiter, 1)[0].isdigit() else dict
```

Comment:
I think we can add a comment explaining what this does and why we need it? 

also, I think this will raise an error if `dt` is `{3: "is an int"}` because `list(dt)[0]` is an int, which doesn't have `split`

Reply:
You are right! This will raise an error in your example.

But I think this function now is only for one purpose: correctly unflatten `config`. And as far as I know all the flattened config is a dict, and its keys are all `str`.
Therefore, your example may not appear. This will be confirmed (or not) by unit test.

Actually I think writing such a function with strong robustness isn't that easy. So I agree with adding a comment explaining what this does, especially its limitation. And perhaps we need to add more test?

Reply:
I've added a comment in the latest commit :)